frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/Model.java

```diff
@@ -116,19 +122,49 @@ public void addFirst(Job job) {
         jobsDb.get(DEFAULT_DATA_QUEUE).addFirst(job);
     }
 
-    public Job nextJob(String threadId) throws InterruptedException {
-        return nextJob(threadId, Long.MAX_VALUE);
-    }
+    public void pollBatch(String threadId, long waitTime, Map<String, Job> jobsRepo)
+            throws InterruptedException {
+        if (jobsRepo == null || threadId == null) {
+            logger.error("Invalid input given provided");
+            throw new IllegalArgumentException();
```

Comment:
You can also pass the appropriate message in the exception perhaps ?

Reply:
Piyush is right, Add the message to exception and remove logging.

Reply:
True. The logging shouldn't be there. Good catch.